protocol-models/src/main/resources/airbyte_protocol/v0/airbyte_protocol.yaml

```diff
@@ -464,6 +464,8 @@ definitions:
       - stream
       - sync_mode
       - destination_sync_mode
+      - current_generation_id
+      - sync_id
```

Comment:
are there any backwards compatibility concerns that we need to worry about by making these required?

Reply:
That's a good point, there will be if a migrated connector is running on a non migrated platform.
@cgardens Is it ok if I move it back to a dedicated object with the 3 fields being required and the new object will be optional in the `ConfiguredAirbyteStream`?

Reply:
my intuition is that we are actually safe, because i can't think of a time that a connector ever gives a configured catalog to the platform. it's only ever the platform giving it to a connector. so maybe it's safe? I'd have to dig through the code a bit more to sanity check it.

Reply:
My concern is when the platform gives a catalog to a connector with those fields. If the connector are migrated, those fields will be require but not present.

Reply:
When would that happen though? If the platform knows about though fields it should also populate them, right?

Reply:
I think that it can happen in OSS where we can have the connectors running on an old platform. The platform won't know about the fields but the connector will do.

Reply:
ah. good point.

Reply:
So does the solution which extract those fields in a dedicated object and make it optional in the configuredAirbyteStream sounds good?

Reply:
i'm not excited about that. I think it's letting a migration details dictate the design. i'd rather make them all nullable but specify that they are nullable for backwards compatibility reasons and if they are null what the default behavior is. 

@evantahler wdyt? relates to your comment below.

Reply:
Not ideal, but I think it's the best we can do.  Added to https://docs.google.com/document/d/1wnnyYT7nzEyrvTiCOQGImfbgCyiTaBU9QIzUmojeozg/edit